ci/test-conformance-aks.sh

```diff
@@ -256,16 +256,21 @@ function run_conformance() {
     echo "=== Running Antrea Conformance and Network Policy Tests ==="
     ${GIT_CHECKOUT_DIR}/ci/run-k8s-e2e-tests.sh --e2e-conformance --e2e-network-policy \
       --kube-conformance-image-version ${KUBE_CONFORMANCE_IMAGE_VERSION} \
-      --log-mode ${MODE} > ${GIT_CHECKOUT_DIR}/aks-test.log
+      --log-mode ${MODE} > ${GIT_CHECKOUT_DIR}/aks-test.log || TEST_FAILURE=true
```

Comment:
nit: if the command on the left fails, we probably won't have a xxx-test.log to look at, so "Failed cases exist" might be a bit misleading for this case?

Reply:
I think the behavior is the same as before in that respect?
But also, in which scenario do you think there will be no *.log file? If some conformance tests fail, `run-k8s-e2e-tests.sh` should still emit the test results to stdout and these test results should show up in the log file.
This PR essentially moves the `grep` command from the cloud-specific test scripts to `run-k8s-e2e-tests.sh` to avoid duplicating it.

Reply:
In practice I've seen something like: 1) a wrong Antrea repo is pulled as param, so `${GIT_CHECKOUT_DIR}/ci/run-k8s-e2e-tests.sh does not exist` 2) there are some issues with Sonobouy installation checks in `run-k8s-e2e-tests.sh`, so no tests are run. In those cases, the original behavior is exit with 1 IIRC, so there's no *.log files archived. But if the same happens here, we'd treat the env failures as TEST_FAILURE.

Reply:
I think that's a good point. I have pushed a new commit and I am not using a different exit code for test failure (1) and internal error in test script (2). I took inspiration from the pytest framework: https://docs.pytest.org/en/documentation-restructure/how-to/usage.html